e2e/tools/validator/src/validator/config/__init__.py

```diff
@@ -57,6 +57,113 @@ def __repr__(self):
         return f"<Config {self.remote}@{self.prometheus}>"
 
 
+# consider switching to dataclass to avoid repeated fields
+class Local(NamedTuple):
+    load_curve: str
+    iterations: str
+    mount_dir: str
+
+
+class LocalProcess(NamedTuple):
+    isolated_cpu: str
+    load_curve: str
+    iterations: str
+    mount_dir: str
+
+
+class LocalContainer(NamedTuple):
+    isolated_cpu: str
+    container_name: str
+    load_curve: str
+    iterations: str
+    mount_dir: str
+
+
+class LocalPrometheus(NamedTuple):
+    url: str
+    rate_interval: str
+    step: str
+    job: str
+
+
+class BMValidator(NamedTuple):
+    log_level: str
+    prom: LocalPrometheus
+    node: Local
+    process: LocalProcess
+    container: LocalContainer
+    validations_file: str
+
+
+def bload(config_file: str) -> BMValidator:
+    """
+    Reads Baremetal YAML configuration file and returns a Config object.
+
+    Args:
+        config_file (str): Path to Baremetal YAML configuration file.
+
+    Returns:
+        BMValidator: A named tuple containing configuration values for Baremetal Validation.
+    """
+    with open(config_file) as file:
+        config = yaml.safe_load(file)
+
+    log_level = config.get("log_level", "warn")
+    prom_config = config["prometheus"]
+    if not prom_config:
+        prom_config = {}
+    prom = LocalPrometheus(
+        url=prom_config.get("url", "http://localhost:9090"),
+        rate_interval=prom_config.get("rate_interval", "20s"),
+        step=prom_config.get("step", "3s"),
+        job=prom_config.get("job", "metal")
+    )
+    print(prom)
+
+    default_config = config["config"]
+    node_config = config["node"]
+    process_config = config["process"]
+    container_config = config["container"]
+    # node config
+    if not node_config:
+        node_config = {}
+    node = Local(
+        load_curve=node_config.get("load_curve", default_config["load_curve"]),
+        iterations=node_config.get("iterations", default_config["iterations"]),
+        mount_dir=os.path.expanduser(node_config.get("mount_dir", default_config["mount_dir"]))
```

Comment:
Why do we need to mount the directory? Are we deploying Kepler on k8s or via docker compose on BM?

Reply:
Currently, kepler is deployed with docker compose for simplicity. however, this should still work the same even in a kubernetes environment. The mount directory is specifically to get the start and end times of the stressor script as a log file from the container. In a non container, the log file will be saved at the mount directory.

Reply:
Shouldn't we target a single environment for now and focus on expanding it further? Currently, we use docker compose for our development and validations. Let's stick with that for now and afterwards add support for k8s 
Wdyt?